garrysmod/lua/includes/extensions/string.lua

```diff
@@ -356,7 +356,7 @@ end
 
 function string.Comma( number, str )
 
-	local replace = str == nil and "%1,%2" or "%1" .. str .. "%2"
+	local replace = ( str == nil or isnumber( str ) ) and "%1,%2" or "%1" .. str .. "%2"

```

Comment:
Yeah I think it'd be better to throw an explicit error rather than fail silently, similarly like it is done here:
https://github.com/Facepunch/garrysmod/blob/9bbd7c8af0dda5bed88e3f09fbdf5d4be7e012f2/garrysmod/lua/includes/extensions/util.lua#L234

Reply:
Made a change, now throwing an explicit error, let me know if the thrown message is acceptable.

Reply:
```
	if ( str ~= nil and !isstring( str ) ) then
		error( "bad argument #2 to 'string.Comma' (string expected, got " .. type( str ) .. ")" )
	end
```
This would be more consistent with other functions.

Reply:
> ```
> 	if ( str ~= nil and !isstring( str ) ) then
> 		error( "bad argument #2 to 'string.Comma' (string expected, got " .. type( str ) .. ")" )
> 	end
> ```
> 
> This would be more consistent with other functions.

Whilst this will stop a true `2` from being passed, my only concern with this if statement, is that it will still allow `"2"` to be passed as `"2"` is a valid string where `isstring` is concerned, however the pattern matching of `gsub` (` ^(-?%d+)(%d%d%d) `) cares not for the data type and will still match on `"2"` and "2" indiscriminately; the infinite loop would still be encountered. We need to explicitly check that it is not a number here to prevent the infinite loop.